From 5403899c6aae0a61a70e4f494cedec4c72ff61a0 Mon Sep 17 00:00:00 2001
From: Guo Jie <guojie@loongson.cn>
Date: Thu, 1 Dec 2022 20:26:54 +0800
Subject: [PATCH 15/28] gcc-8-vec: Enable -fsched-pressure by default when
 optimizing.

gcc/ChangeLog:
	* common/config/loongarch/loongarch-common.c: The default algorithm is SCHED_PRESSURE_WEIGHTED.
	* testsuite/gcc.target/loongarch/cmov_ii.c: Fix the selector's rule.

Change-Id: I7c40a7e3537bc2397b30bce292f0e34d37613706
---
 src/gcc/common/config/loongarch/loongarch-common.c | 2 ++
 src/gcc/testsuite/gcc.target/loongarch/cmov_ii.c   | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/gcc/common/config/loongarch/loongarch-common.c b/src/gcc/common/config/loongarch/loongarch-common.c
index bdc5b939c..ccdc8f498 100644
--- a/src/gcc/common/config/loongarch/loongarch-common.c
+++ b/src/gcc/common/config/loongarch/loongarch-common.c
@@ -34,6 +34,8 @@ along with GCC; see the file COPYING3.  If not see
 static const struct default_options loongarch_option_optimization_table[] =
 {
   { OPT_LEVELS_ALL, OPT_fasynchronous_unwind_tables, NULL, 1 },
+  /* Enable -fsched-pressure by default when optimizing.  */
+  { OPT_LEVELS_1_PLUS, OPT_fsched_pressure, NULL, 1 },
   { OPT_LEVELS_NONE, 0, NULL, 0 }
 };
 
diff --git a/src/gcc/testsuite/gcc.target/loongarch/cmov_ii.c b/src/gcc/testsuite/gcc.target/loongarch/cmov_ii.c
index 66f656fd6..6847ae208 100644
--- a/src/gcc/testsuite/gcc.target/loongarch/cmov_ii.c
+++ b/src/gcc/testsuite/gcc.target/loongarch/cmov_ii.c
@@ -1,7 +1,7 @@
 /* Test asm const. */
 /* { dg-do compile } */
 /* { dg-options "-O2 -S" } */
-/* { dg-final { scan-assembler-times "main:.*xor.*masknez.*maskeqz.*or.*" 1 } } */
+/* { dg-final { scan-assembler-times "main:.*xor.*masknez.*maskeqz.*or.*\|main:.*xor.*maskeqz.*masknez.*or.*" 1 } } */
 #include <stdio.h>
 
 extern void foo_ii(int*, int*, int*, int*);
-- 
2.31.1

